app/src/main/java/com/jerboa/ui/components/comment/edit/CommentEditActivity.kt

```diff
@@ -24,29 +24,24 @@ import com.jerboa.model.AccountViewModel
 import com.jerboa.model.CommentEditViewModel
 import com.jerboa.ui.components.common.JerboaSnackbarHost
 import com.jerboa.ui.components.common.getCurrentAccount
-import com.jerboa.util.InitializeRoute
 
 object CommentEditReturn {
     const val COMMENT_VIEW = "comment-edit::return(comment-view)"
+    const val COMMENT_SEND = "comment-edit::send(comment-view)"
 }
 
 @Composable
 fun CommentEditActivity(
-    commentView: CommentView,
     appState: JerboaAppState,
     accountViewModel: AccountViewModel,
 ) {
     Log.d("jerboa", "got to comment edit activity")
 
     val account = getCurrentAccount(accountViewModel = accountViewModel)
-    val snackbarHostState = remember {
-        SnackbarHostState()
-    }
+    val snackbarHostState = remember { SnackbarHostState() }
 
     val commentEditViewModel: CommentEditViewModel = viewModel()
-    InitializeRoute(commentEditViewModel) {
-        commentEditViewModel.initialize(commentView)
-    }
+    val commentView = appState.getPrevReturn<CommentView>(key = CommentEditReturn.COMMENT_SEND)
```

Comment:
This seems like the heart of it here, just an alternate way to send initial data, and remove it from the model.

Reply:
Even before Jetpack compose passing data between activities was never really done cleanly.

Take a look at IfR does, it uses intents to pass data between activities and Preferences API for persistence

https://github.com/Docile-Alligator/Infinity-For-Reddit/blob/02278263579d76f88eef9c643bd7ace412f99985/app/src/main/java/ml/docilealligator/infinityforreddit/activities/SearchActivity.java#L65-L75

Yeah basically. The big difference here, compared to consumeReturn, is that it never removes the data from the route, it only gets removed when the screen gets popped from backstack.